internal/worker/jwtparser/worker.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jwtparser
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/juju/worker/v3"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/internal/jwtparser"
+)
+
+type jwtParserWorker struct {
+	tomb      tomb.Tomb
+	jwtParser *jwtparser.Parser
+}
+
+// ControllerConfig defines an interface to retrieve controller config.
+type ControllerConfig interface {
+	ControllerConfig() (controller.Config, error)
+}
+
+// HTTPClient defines an interface for an HTTP client
+// with a single GET method.
+type HTTPClient interface {
+	jwk.HTTPClient
+}
+
+// NewWorker returns a worker that provides a JWTParser.
+func NewWorker(configGetter ControllerConfig, httpClient HTTPClient) (worker.Worker, error) {
+	controllerConfig, err := configGetter.ControllerConfig()
+	if err != nil {
+		return nil, errors.Annotate(err, "cannot fetch the controller config")
+	}
+	jwtRefreshURL := controllerConfig.LoginTokenRefreshURL()
+
+	ctx, done := context.WithCancel(context.Background())
+	jwtParser := jwtparser.NewParserWithHTTPClient(ctx, httpClient)
+	if jwtRefreshURL != "" {
+		if err := jwtParser.SetJWKSCache(context.Background(), jwtRefreshURL); err != nil {
```

Comment:
why do we use `context.Background()` here?

Reply:
There is no traditional context available to workers and the jwks cache uses a context as a signalling mechanism to kill the go routines it spawns. So we just need a context that we stop when the worker dies to avoid leaking go routines.

Reply:
I understand that you create context with cancel a few lines above, but why don't you use that context in the SetJWKSCache call?

Reply:
Jeez, that was a mistake >.> thanks for that